internal/pkg/core/composable/providers.go

```diff
@@ -19,6 +19,12 @@ type FetchContextProvider interface {
 type ContextProviderComm interface {
 	context.Context
 
+	// Signal signals that something has changed in the provider.
+	//
+	// Note: This should only be used by fetch context providers, standard context
+	// providers should use Set to update the overall state.
+	Signal()
```

Comment:
I want to make sure I understand why fetch providers shouldn't call set.

Is this because `Set` is meant to add a mapping and its value, where as a `Fetch` provider is supposed to be provided the list of keys to get?

In the context of the k8s secrets provider we wouldn't call `Set` because we aren't adding secrets keys to the policy, the policy is using the provider to set/fetch their value?

Reply:
I think you are saying it correctly, but the working of set/fetch might be a little off.

`Set` should not be called by the k8s secrets provider because the available keys are not known in advanced like all the other providers. That is the main reason the k8s secrets provider is a fetch provider, because we don't want it to load every possible secret. Because of this reason it needs a way to just signal that something has changed, and then `Fetch()` will be called again as the Elastic Agent processes the variables in the configuration.